### PR TITLE
feat/change allocate investigator to investigation footer actions

### PR DESCRIPTION
**Pull request by ItayBenMoshe**
[1246](https://dev.azure.com/spectrumFactory/CoronaI/_workitems/edit/1246/)

### CODE REVIEW

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx [hunk 1]
```diff
@@ -36,39 +36,36 @@ const multipleAssignments = 'הקצאות';
...
     const [openGroupedInvestigations, setOpenGroupedInvestigations] = useState<boolean>(false);
 
     const {
         handleOpenDesksDialog,
         handleCloseDesksDialog,
-        handleOpenInvestigatorsDialog,
-        handleCloseInvestigatorsDialog,
         handleOpenGroupedInvestigations,
         handleCloseGroupedInvestigations,
+        handleOpenInvesigatorAloocationDialog,
```
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx:53]: allocate *

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx [hunk 2]
```diff
@@ -36,39 +36,36 @@ const multipleAssignments = 'הקצאות';
...
+        handleCloseInvesigatorAloocationDialog,
```
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx:54]: same

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx [hunk 3]
```diff
@@ -129,7 +129,7 @@ const InvestigationTableFooter: React.FC<Props> = React.forwardRef((props: Props
             icon: PersonPin,
             displayTitle: `${isSingleInvestigation ? singleAssignment : multipleAssignments} לחוקר`,
             errorMessage: '',
-            onClick: handleOpenInvestigatorsDialog
+            onClick: handleOpenInvesigatorAloocationDialog
```
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx:132]: same

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts [hunk 4]
```diff
@@ -52,6 +51,13 @@ const useInvestigationTableFooter = (parameters: InvestigationTableFooterParamet
         onDialogClose();
     }
 
+    const handleOpenInvesigatorAloocationDialog = () => setIsInvestigatorAllocationDialogOpen(true);
```
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts:54]: same

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts [hunk 5]
```diff
@@ -52,6 +51,13 @@ const useInvestigationTableFooter = (parameters: InvestigationTableFooterParamet
...
+
+    const handleCloseInvesigatorAloocationDialog = () => {
```
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts:56]: same

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts [hunk 6]
```diff
@@ -172,13 +146,12 @@ const useInvestigationTableFooter = (parameters: InvestigationTableFooterParamet
     return {
         handleOpenDesksDialog,
         handleCloseDesksDialog,
-        handleOpenInvestigatorsDialog,
-        handleCloseInvestigatorsDialog,
+        handleOpenInvesigatorAloocationDialog,
+        handleCloseInvesigatorAloocationDialog,
```
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts:150]: allocation

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts [hunk 7]
```diff
@@ -172,13 +146,12 @@ const useInvestigationTableFooter = (parameters: InvestigationTableFooterParamet
     return {
         handleOpenDesksDialog,
         handleCloseDesksDialog,
-        handleOpenInvestigatorsDialog,
-        handleCloseInvestigatorsDialog,
+        handleOpenInvesigatorAloocationDialog,
```
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts:149]: allocation

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooterInterfaces.ts [hunk 8]
```diff
@@ -1,32 +1,28 @@
...
 
 export interface InvestigationTableFooterOutcome {
     handleOpenDesksDialog: () => void;
     handleCloseDesksDialog: () => void;
-    handleOpenInvestigatorsDialog: () => void;
-    handleCloseInvestigatorsDialog: () => void;
     handleOpenGroupedInvestigations: () => void;
     handleCloseGroupedInvestigations: () => void;
+    handleOpenInvesigatorAloocationDialog: () => void;
+    handleCloseInvesigatorAloocationDialog: () => void;
```
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooterInterfaces.ts:24]: allocation

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooterInterfaces.ts [hunk 9]
```diff
@@ -1,32 +1,28 @@
...
 }
 
 export interface InvestigationTableFooterOutcome {
     handleOpenDesksDialog: () => void;
     handleCloseDesksDialog: () => void;
-    handleOpenInvestigatorsDialog: () => void;
-    handleCloseInvestigatorsDialog: () => void;
     handleOpenGroupedInvestigations: () => void;
     handleCloseGroupedInvestigations: () => void;
+    handleOpenInvesigatorAloocationDialog: () => void;
```
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooterInterfaces.ts:23]: allocation

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx [hunk 10]
```diff
@@ -160,11 +160,13 @@ const InvestigationTableFooter: React.FC<Props> = React.forwardRef((props: Props
...
                 allCounties={allCounties}
             />
-            <TransferInvestigationInvestigator
-                open={openInvestigatorsDialog}
-                onConfirm={handleConfirmInvestigatorsDialog}
-                onClose={handleCloseInvestigatorsDialog}
-                allInvestigators={allInvestigators}
+            <InvestigatorAllocationDialog
+                isOpen={isInvestigatorAllocationDialogOpen}
+                handleCloseDialog={handleCloseInvesigatorAloocationDialog}
```
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx:165]: allocation

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx (hunk 3 above)
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx:132]: allocation

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx (hunk 2 above)
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx:54]: allocation

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx (hunk 1 above)
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx:53]: allocation

**Review by NaamanTsur — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx [hunk 11]
```diff
@@ -207,24 +210,22 @@ const InvestigationTable: React.FC = (): JSX.Element => {
                     />
                 );
             case TableHeadersNames.investigatorName:
-                if (selectedRow === indexedRow.epidemiologyNumber && shouldOpenInvestigatorAllocation) {
+                if (selectedRow === indexedRow.epidemiologyNumber && isOpenInvestigatorAllocation) {
                     return (
                         <InvestigatorAllocationDialog
```
[the remark this answers is not in the document]
Comment (NaamanTsur) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx:215]: If you click the investigator cell you can see the name disappearing before the dialog opens.

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigatorAllocation/InvestigatorAllocationCell.tsx [hunk 12]
```diff
@@ -8,7 +8,9 @@ import useStyles from './InvestigatorAllocationCellStyles';
 
 const InvestigatorAllocationCell: React.FC<Props> = (props: Props) => {
 
-    const { investigatorName, epidemiologyNumber, epiNumOnInvestigatorNameHover, setEpiNumOnInvestigatorNameHover, } = props;
+    const { investigatorName, epidemiologyNumber } = props;
+
+    const [epiNumOnInvestigatorNameHover, setEpiNumOnInvestigatorNameHover] = useState<number>(defaultEpidemiologyNumber);
```
[the remark this answers is not in the document]
Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigatorAllocation/InvestigatorAllocationCell.tsx]: fixed

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts (hunk 4 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts]: fixed

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts (hunk 5 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts]: fixed

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts (hunk 6 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts]: fixed

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts (hunk 7 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/useInvestigationTableFooter.ts]: fixed

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooterInterfaces.ts (hunk 8 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooterInterfaces.ts]: fixed

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooterInterfaces.ts (hunk 9 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooterInterfaces.ts]: fixed

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx (hunk 10 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx]: fixed

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx (hunk 3 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx]: fixed

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx (hunk 2 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx]: fixed

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx (hunk 1 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/InvestigationTableFooter.tsx]: fixed

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx [hunk 13]
```diff
@@ -112,8 +112,7 @@ const InvestigationTable: React.FC = (): JSX.Element => {
     const [allGroupedInvestigations, setAllGroupedInvestigations] = useState<Map<string, InvestigationTableRow[]>>(new Map());
     const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
     const [shouldOpenPopover, setShouldOpenPopover] = useState<boolean>(false);
-    const [shouldOpenInvestigatorAllocation, setShouldOpenInvestigatorsAllocation] = useState<boolean>(false);
-    const [epiNumOnInvestigatorNameHover, setEpiNumOnInvestigatorNameHover] = useState<number>(defaultEpidemiologyNumber);
+    const [isOpenInvestigatorAllocation, setIsOpenInvestigatorsAllocation] = useState<boolean>(false);
```
[the remark this answers is not in the document]
Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx]: fixed

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx]: fixed

**Review by nevobonneh — DISMISSED** [review]

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx [hunk 14]
```diff
@@ -207,27 +210,23 @@ const InvestigationTable: React.FC = (): JSX.Element => {
...
-                } else {
                     return (
-                        <InvestigatorAllocationCell
-                            investigatorName={indexedRow[cellName as keyof typeof TableHeadersNames]}
-                            epidemiologyNumber={indexedRow.epidemiologyNumber}
-                            epiNumOnInvestigatorNameHover={epiNumOnInvestigatorNameHover}
-                            setEpiNumOnInvestigatorNameHover={setEpiNumOnInvestigatorNameHover}
-                        />
+                        <>
+                            { selectedRow === indexedRow.epidemiologyNumber && isInvestigatorAllocationDialogOpen &&
```
Comment (NaamanTsur) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx:215]: I meant that you can put the InvestigatorAllocationDialog outside of this function in the InvestigationTable render area. 
Right now you have a condition for rendering the dialog but you also have an `isOpen` property which hides it. 
If you remove the condition for rendering the dialog you have a dialog in the dom for each cell in the table. 
If there are 100 investigations there will be 100 dialogs hidden in the dom.

**Review by NaamanTsur — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx (hunk 14 above)
Comment (NaamanTsur) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx:215]: We should have one of this dialog for all the investigations in the table

**Review by NaamanTsur — APPROVED** [review]

**Review by nevobonneh — APPROVED** [review]

**Pull request merged**